### ISSUE TITLE
Variable doesn't work in included templates if isn't used anywhere after in parent template

### ISSUE THREAD

**Issue by vlad1777d** [pallets/jinja#743]
Hello.

Sorry me please, maybe I understood wrong the documentation, or the problem is in Pelican static website generator, which I use. Also sorry that I had not asked about it on stackoverflow and that I paste images instead of templates. I's because of lack of time, If I'll not do it - I'll forget about it and will do never. Hope, it will help to correct bug in Jinja2, if it's present.

## Expected Behavior
If I declared variable in parent template, it must be available in all included template.

## Actual Behavior
I declared variable in parent template, it's available in all included template only in case, when this variable was used anywhere in parent template, but lower, that that include statements are placed.

## Template Code
Base template:
![screenshot from 2017-07-13 03-08-13](https://user-images.githubusercontent.com/12682937/28145173-d2526782-6778-11e7-8302-20fd6dfd5dfb.png)
Here variable was declared on line 2.

Child template:
![screenshot from 2017-07-13 03-08-35](https://user-images.githubusercontent.com/12682937/28145193-e88b331c-6778-11e7-8a19-b819eed313f8.png)

Now "root_relative" would not be rendered. But it would be rendered in case, when I'll place this variable into parent template somewhere below than include statements:
![screenshot from 2017-07-13 03-09-11](https://user-images.githubusercontent.com/12682937/28145215-1eada024-6779-11e7-9a24-b4d3563a1415.png)

## Full Traceback
There were no errors in logs when I called rendering of my website with:
`pelican`, `pelican --debug`, `pelican --vebrose`

## Your Environment
* Python version: 2.7.12 (also I have 3.5.2, but Pelican uses 2.7.12)
* Jinja version: 2.8-1 (from Linux Mint's repository)
* OS: Linux Mint 18.2
* Pelican: 3.6.3-1

Sorry another time for I didn't checked out more about this behavior and didn't asked about it on Stackoverflow. It's because of lack of time.

Best regards, Vladislav.

**Comment by davidism** [comment]
Please paste code, not pictures of code.

**Comment by vlad1777d** [comment]
@davidism , made some example.

Template a.html:
```
{% set var_a = "variable_text" %}
<html>
<head>
	<title>{{ SITENAME }}</title>
</head>


<body>

	{% include "c.html" %}
	{{ var_a }}

</body>
</html>
```


Template b.html:
```
{% set var_a = "variable_text" %}
<html>
<head>
	<title>{{ SITENAME }}</title>
</head>


<body>

	{% include "c.html" %}

</body>
</html>
```

Template c.html:
```
This is template c.html. Here must be variable: {{ var_a }}. Template finished.
```




Output a.html:
```
<html>
<head>
	<title>Bug example</title>
</head>


<body>

This is template c.html. Here must be variable: variable_text. Template finished.	variable_text

</body>
</html>
```



Output b.html:
```
<html>
<head>
	<title>Bug example</title>
</head>


<body>

This is template c.html. Here must be variable: variable_text. Template finished.
</body>
</html>
```

Here is pelican's project archive:
[jinja-bug-example.tar.gz](https://github.com/pallets/jinja/files/1148010/jinja-bug-example.tar.gz)

Source templates are located in ./sources/theme_general/templates
Output templates are located in ./output

To build project you need **pelican** package (apt install pelican).
Then open root folder of a project in terminal and write "pelican".

**Comment by vlad1777d** [comment]
@davidism , sorry for disturbing, had you seen my additions to related to this bug ?

**Comment by ricky-undeadcoders** [comment]
I can't reproduce this in version 2.10. It looks to have been solved with #619. 

Can this be closed?
